pkgs/by-name/pdfding/frontend.nix

```diff
@@ -0,0 +1,80 @@
+{
+  stdenv,
+  nodejs,
+  npmHooks,
+  fetchpatch2,
+  fetchNpmDeps,
+  fetchzip,
+  fetchFromGitHub,
+  tailwindcss_4,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pdfding-frontend";
+  version = "1.4.1";
+  src = fetchFromGitHub {
+    owner = "mrmn2";
+    repo = "PdfDing";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-8e80gMdB6U3977dIU7bIAAEguYmi+AWQgUgYPDLCYLI=";
+  };
+
+  npmDeps = fetchNpmDeps {
+    inherit (finalAttrs) src;
+    name = "pdfding-frontend-${finalAttrs.version}-npm-deps";
+    hash = "sha256-v1NFqDnFcRK8sd0bV3ck+LLMYQ90Dl1R1OnBTwWUVUg=";
+  };
+
+  patches = [
+    # remove patch in 1.4.2
+    # package.json has missing version and name
+    (fetchpatch2 {
+      url = "https://github.com/mrmn2/PdfDing/pull/203.patch?full_index=1";
+      hash = "sha256-lKtpqKdyoGZdU4fTegto+YUIduIWbM82RQU9459NpC0=";
+    })
+  ];
+
+  passthru = {
+    pdfjsVersion = "5.4.296"; # see update script
+    pdfjsHash = "sha256-b4W7wETq2CIZm2rJCmXEYvPhQtCbXS76L7GDvng6wn4=";
+    pdfjs = fetchzip {
+      url = "https://github.com/mozilla/pdf.js/releases/download/v${finalAttrs.passthru.pdfjsVersion}/pdfjs-${finalAttrs.passthru.pdfjsVersion}-dist.zip";
+      hash = finalAttrs.passthru.pdfjsHash;
+      stripRoot = false;
+      postFetch = ''
+        rm -rf $out/web/locale \
+        $out/web/standard_fonts \
+        $out/web/compressed.tracemonkey-pldi-09.pdf
+
+        # remove source maps
+        find "$out" -name '*.map' -exec rm -f '{}' \;
+      '';
+    };
```

Comment:
What about moving `pdfjs` to separate package ?

Reply:
I don't know if we would see much use of it in other packages. Also we have pdfjs's version tied to what pdfding specifies it the dockerfile.

Reply:
I agree with moving `passthru` down, but I think we keep `pdfjs` there because it's easier to override with `finalAttrs`.